licensing/src/main/java/io/crate/license/LicenseService.java

```diff
@@ -129,6 +122,81 @@ public void onFailure(Exception e) {
         }
     }
 
+    //@VisibleForTesting
+    boolean verifyLicense(LicenseKey licenseKey) {
+        try {
+            DecodedLicense decodedLicense = decodeLicense(licenseKey);
+            DecryptedLicenseData licenseInfo = decryptLicenseContent(decodedLicense.encryptedContent());
+
+            if (licenseInfo.isExpired()) {
+                //LOGGER.warn("Unable to validate [{}] as it is expired", license);
+                return false;
+            }
+            if (decodedLicense.type() == LicenseType.ENTERPRISE) {
+                return verifySignature(licenseInfo);
```

Comment:
To me it feels a bit redundant to encrypt the content and have a signature as well but maybe I'm missing something. The content we would encrypt with the private key could be unencrypted with the public key we ship and I think that's the only thing that should differ from the self generated license processing. 

Maybe I am missing something though. @mfussenegger what do you think?

Reply:
It's a bit difficult to see the full license verification flow with just the bits and pieces in the diffs.

I guess it would make sense to have some high level javadocs that illustrates the flow?

Reply:
added the high level javadocs.

It is true that in the presence of the signature (enterprise licenses), encryption feels a bit redundant, however I decided to keep it as 
a) it's not that expensive (symmetric)
b) it makes handling the two types of licenses more consistent -> cleaner code

but I'm open to suggestions :)